src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -169,6 +169,8 @@ module Crystal
     end
 
     def visit(node : Path)
+      return if node.type?
```

Comment:
Is this necessary?

Reply:
Yes because it might be analyzed twice now that Path is solved before expansion and after expansion.